backend/flow_api/flow/shared/hook_email_persist_verified_status.go

```diff
@@ -14,62 +16,93 @@ type EmailPersistVerifiedStatus struct {
 func (h EmailPersistVerifiedStatus) Execute(c flowpilot.HookExecutionContext) error {
 	deps := h.GetDeps(c)
 
-	if !deps.Cfg.Emails.RequireVerification {
-		return nil
+	if !c.Stash().Get("email_verified").Bool() {
+		return errors.New("no email has been confirmed as verified")
+	}
```

Comment:
When emailVerification is disabled and I want to verify my email in the profile I get an error when executing the actions `resend_passcode` or `back` or when I enter an invalid passcode e.g. `123456`
```
Code: technical_error, Message: Something went wrong., Cause: the action failed to handle the request: error while executing after hook actions: failed to execute hook action after state: no email has been confirmed as verified
```